app/api/search/route.ts

```diff
@@ -0,0 +1,37 @@
+import { NextRequest } from "next/server";
+import { cookies } from "next/headers";
+import { createClient } from "@/utils/supabase/server";
+
+interface SearchQuery {
+  query: string;
+}
+export async function POST(req: NextRequest) {
+  const cookieStore = cookies();
+  const supabase = createClient(cookieStore);
+
+  const { query }: SearchQuery = await req.json();
+  try {
+    const { data, error } = await supabase
+      .from("posts")
+      .select("post_id, title, description")
+      .like("title", "%" + query + "%");
+
+    if (error) {
+      return new Response(JSON.stringify({ error: error.message }), {
+        status: 500,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+    }
+
+    return new Response(JSON.stringify(data), {
+      status: 200,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  } catch (error) {
+    console.error("error THROW from search", error);
```

Comment:
do you want to keep these `console.log` in, rather than return an error response?